lib/msf/ui/console/command_dispatcher/db/klist.rb

```diff
@@ -15,34 +15,44 @@ def cmd_klist_tabs(str, words)
 
   def cmd_klist_help
     print_line 'List Kerberos tickets in the database'
-    print_line 'Usage: klist [options]'
+    print_line 'Usage: klist [options] [hosts]'
     print_line
     print @@klist_opts.usage
     print_line
   end
 
   @@klist_opts = Rex::Parser::Arguments.new(
     ['-v', '--verbose'] => [false, 'Verbose output'],
-    ['-d', '--delete'] => [ false, 'Delete *all* matching kerberos entries' ],
-    ['-h', '--help'] => [false, 'Help banner']
+    ['-d', '--delete'] => [ false, 'Delete *all* matching kerberos entries'],
+    ['-h', '--help'] => [false, 'Help banner'],
+    ['-i', '--index'] => [true, 'Kerberos entry ID(s) to search for, e.g. `-i 1` or `-i 1,2,3` or `-i 1 -i 2 -i 3`'],
+    ['-a', '--activate'] => [false, 'Activates *all* matching kerberos entries'],
+    ['-A', '--deactivate'] => [false, 'Deactivates *all* matching kerberos entries']
   )
 
   def cmd_klist(*args)
     return unless active?
 
-    delete_count = 0
+    entries_affected = 0
     mode = :list
     host_ranges = []
+    id_search = []
     verbose = false
     @@klist_opts.parse(args) do |opt, _idx, val|
       case opt
       when '-h', '--help'
         cmd_klist_help
         return
-      when '-v', '--vebose'
+      when '-v', '--verbose'
         verbose = true
       when '-d', '--delete'
         mode = :delete
+      when '-i', '--id'
+        id_search.concat(val.split(/,\s*|\s+/)).uniq # allows 1 or 1,2,3 or "1 2 3" or "1, 2, 3"
```

Comment:
Original suggested was:

```suggestion
        id_search.concat(val.split(/,\s*|\s+/)).uniq # allows 1 or 1,2,3 or "1 2 3" or "1, 2, 3"
```

as `uniq` creates a new array and doesn't mutate the original:

```
>> a = [1, 2, 2]
=> [1, 2, 2]
>> a.uniq
=> [1, 2]
>> a
=> [1, 2, 2]
```

Reply:
my bad sorted now